integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/NessieSparkSqlExtensionsParser.scala

```diff
@@ -106,86 +115,130 @@ class NessieSparkSqlExtensionsParser(delegate: ParserInterface)
+          )
+        case CommandType.DROP_REFERENCE =>
+          val spec = commandSpec.asInstanceOf[DropReferenceCommandSpec]
+          return DropReferenceCommand(
+            spec.getRef,
+            spec.getRefType.equals("BRANCH"),
+            Option(spec.getInCatalog),
+            !spec.isConditional
+          )
+        case CommandType.ASSIGN_REFERENCE =>
+          val spec = commandSpec.asInstanceOf[AssignReferenceCommandSpec]
+          return AssignReferenceCommand(
+            spec.getRef,
+            spec.getRefType.equals("BRANCH"),
+            Option(spec.getTo),
+            Option(spec.getRefTimestampOrHash),
+            Option(spec.getInCatalog)
+          )
+        case CommandType.LIST_REFERENCES =>
+          val spec = commandSpec.asInstanceOf[ListReferencesCommandSpec]
+          return ListReferenceCommand(
+            Option(spec.getFilter),
+            Option(spec.getStartsWith),
+            Option(spec.getContains),
+            Option(spec.getInCatalog)
+          )
+        case CommandType.MERGE_BRANCH =>
+          val spec = commandSpec.asInstanceOf[MergeBranchCommandSpec]
+          return MergeBranchCommand(
+            Option(spec.getRef),
+            Option(spec.getRefTimestampOrHash),
+            Option(spec.getInto),
+            spec.isDryRun,
+            Option(spec.getDefaultMergeBehavior),
+            spec.getKeyMergeBehaviors,
+            Option(spec.getInCatalog)
+          )
+        case CommandType.SHOW_LOG =>
+          val spec = commandSpec.asInstanceOf[ShowLogCommandSpec]
+          return ShowLogCommand(
+            Option(spec.getRef),
+            Option(spec.getRefTimestampOrHash),
+            Option(spec.getLimit).flatMap(l => if (l == 0) None else Option(l)),
+            Option(spec.getInCatalog)
+          )
+        case CommandType.SHOW_REFERENCE =>
+          val spec = commandSpec.asInstanceOf[ShowReferenceCommandSpec]
+          return ShowReferenceCommand(
+            Option(spec.getRef),
+            Option(spec.getRefTimestampOrHash),
+            Option(spec.getInCatalog)
+          )
+        // case CommandType.SHOW_CONTENT   => ???
+        // case CommandType.REVERT_CONTENT => ???
+        // case CommandType.CREATE_NAMESPACE => ???
+        // case CommandType.DROP_CONTENT => ???
+        // case CommandType.ALTER_NAMESPACE => ???
+        // case CommandType.LIST_CONTENTS   => ???
+        case _ => delegate.parsePlan(sqlText)
       }
+
     } catch {
-      case e: NessieParseException if e.command.isDefined =>
-        throw e
-      case e: NessieParseException =>
-        throw e.withCommand(command)
-      case e: AnalysisException =>
-        val position = Origin(e.line, e.startPosition)
-        throw new NessieParseException(
-          Option(command),
-          e.message,
-          position,
-          position
-        )
+      case _: ParseException =>
+      // TODO check whether it's a Nessie statement
```

Comment:
Not throwing anything?

Reply:
Indirect, via the call below. We'd need some `isNessieStatement` to do something "proper"